builder/amazon/ebssurrogate/step_snapshot_volumes.go

```diff
@@ -104,11 +108,13 @@ func (s *StepSnapshotVolumes) Cleanup(state multistep.StateBag) {
 		ec2conn := state.Get("ec2").(*ec2.EC2)
 		ui := state.Get("ui").(packer.Ui)
 		ui.Say("Removing snapshots since we cancelled or halted...")
+		s.snapshotMutex.RLock() // read lock
 		for _, snapshotId := range s.snapshotIds {
 			_, err := ec2conn.DeleteSnapshot(&ec2.DeleteSnapshotInput{SnapshotId: &snapshotId})
 			if err != nil {
 				ui.Error(fmt.Sprintf("Error: %s", err))
 			}
 		}
+		s.snapshotMutex.Unlock()
```

Comment:
Again I would recommend not using rwlock mutexes but if you think this is better this needs to be changed to 
```suggestion
		s.snapshotMutex.Runlock()
```